soc/arm64/ti_sitara/am6x/mmu_regions.c

```diff
@@ -19,6 +22,11 @@ static const struct arm_mmu_region mmu_regions[] = {
 			      DT_REG_ADDR_BY_IDX(DT_NODELABEL(gic), 1),
 			      DT_REG_SIZE_BY_IDX(DT_NODELABEL(gic), 1),
 			      MT_DEVICE_nGnRnE | MT_P_RW_U_NA | MT_NS),
+
+	MMU_REGION_FLAT_ENTRY("PINCTRL",
+			      DT_REG_ADDR(DT_NODELABEL(pinctrl)),
+			      ROUND_UP_MMU_PAGE(DT_REG_SIZE(DT_NODELABEL(pinctrl))),
+			      MT_DEVICE_nGnRnE | MT_P_RW_U_NA | MT_NS),
```

Comment:
No, we are trying to move away from forcing a 1:1 mapping. The pinctrl should use the device MMIO APIs to dynamically map the memory region, see for example how this is done for `drivers/serial/uart_pl011.c`